wireguard-go: 0.0.20230223 -> 0-unstable-2023-12-11; unbreak

winterqt opened the pull request:
9e2f3860220280a5630971478b53c8ad9a991ca8 [0] fixed the compilation on Go >1.22, but there are a bunch of fixes after the latest tag that I think make sense to pull in at this point.

[0]: https://git.zx2c4.com/wireguard-go/commit/?id=9e2f3860220280a5630971478b53c8ad9a991ca8

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

winterqt committed:
wireguard-go: 0.0.20230223 -> 0-unstable-2023-12-11; unbreak

9e2f3860220280a5630971478b53c8ad9a991ca8 [0] fixed the compilation on
Go >1.22, but there are a bunch of fixes after the latest tag that
I think make sense to pull in at this point.

[0]: https://git.zx2c4.com/wireguard-go/commit/?id=9e2f3860220280a5630971478b53c8ad9a991ca8

winterqt commented on pkgs/tools/networking/wireguard-go/default.nix:
```diff
@@ -2,25 +2,25 @@
 buildGoModule rec {
   pname = "wireguard-go";
-  version = "0.0.20230223";
+  version = "0-unstable-2023-11-10";
```
Note: this commit has an author date of November, but was committed in December. Do we prefer a specific date?

RossComputerGuy replied:
Probably committed date is best to go with since that's when it was actually "brought in".

katexochen replied:
afaik commit date is what matters, but the language in the docs isn't totally clear: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md?plain=1#L391

winterqt replied:
Yeah, could be worth some clarification. I'll switch it to commit date though (I was personally going back and forth), thanks for the insight!

winterqt committed:
wireguard-go: move to by-name

winterqt committed:
wireguard-go: actually run tests

None of the tests are in the root directory (except for the code
formatting one), so they were never actually run until now.

winterqt committed:
wireguard-go: add winter to maintainers

The current maintainers seem to be inactive, and I'm willing to step in
and keep this package alive. :)

winterqt commented:
Figured I'd make some other additions while I was at it. :)

katexochen commented on pkgs/by-name/wi/wireguard-go/package.nix:
```diff
@@ -0,0 +1,77 @@
+
+    ldflags = [
+      "-s"
+      "-w"
```
```suggestion
```

implied by -s

winterqt committed:
wireguard-go: drop redundant ldflag

winterqt commented:
Should be good now. (Sorry for nagging -- I'd merge myself but figured I'd make sure there was nothing else you had comments on.)
